lib/fetch/headers.js

```diff
@@ -308,24 +280,28 @@ class Headers {
     return this[kHeadersList].set(String(args[0]), String(args[1]))
   }
 
+  // The value pairs to iterate over are the return value of running sort
+  // and combine with this’s header list.
   * keys () {
-    const clone = this[kHeadersList].slice()
-    for (let index = 0; index < clone.length; index += 2) {
-      yield clone[index]
+    const listFreezed = new HeadersList([...this[kHeadersList]].sort())
+    for (const key of listFreezed.keys()) {
+      yield key
```

Comment:
No point in converting back into headers list. Also I don't think the sort works here...